coriolis/osmorphing/windows.py

```diff
@@ -68,6 +69,39 @@
 REQUIRED_DETECTED_WINDOWS_OS_FIELDS = [
     "version_number", "edition_id", "installation_type", "product_name"]
 
+INTERFACES_PATH_FORMAT = (
+    "HKLM:\\%s\\ControlSet001\\Services\\Tcpip\\Parameters\\Interfaces")
+
+STATIC_IP_SCRIPT_TEMPLATE = """
+$NICS_INFO = ConvertFrom-Json "%(nics_info)s"
+$IPS_INFO = ConvertFrom-Json "%(ips_info)s"
+
+$i = 0
+foreach ($nic in $NICS_INFO) {
+    $mac_address = $nic.mac_address.replace(':', '-').toUpper()
+    $interface = Get-NetAdapter | Where MacAddress -eq $mac_address
+    if (-Not $interface) {
+        $interface = @(Get-NetAdapter | Where Status -eq Up)[$i]
```

Comment:
I don't get this part, is this assuming that the new NIC devices will always be in order on the new machine? What is said order based on? What guarantees do we have that it will be consistent?

Ideally please write a comment somewhere in this script for posterity.

Reply:
This is the proposed backdoor logic. If we can't find the NIC by MAC address, the interface will be selected by the nics_info order.

Reply:
This unfortunately has the potential to screw things up harder when our assumption is wrong than any "gains" we get from it being right.

We could hypothetically add better "identification logic" in the future (by exporting more info from the source platform like source network range/gateway) to make more educated matches.

Until then, I'd suggest we simply drop this and just log out somehwere that no matching interface was found for the MAC address (requiring MACs to be preserved seems like a more than reasonable enough requirement IMO)